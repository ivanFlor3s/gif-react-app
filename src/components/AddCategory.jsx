import React, { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategory = ({addGifHandler}) => {
  const [inputValue, setInputValue] = useState("");

  const changeHandler = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
      e.preventDefault()

      addGifHandler((categorias = []) => {
          if(inputValue.trim().length < 3){
              return [...categorias]
          }
          else if(!categorias.includes(inputValue)){
            setInputValue('')
            return [...categorias,inputValue]
          }
          else{
              console.error('Ya se agrego la categoria', inputValue)
              return [...categorias]
          }
        })
  }

  return (
    <form onSubmit={handleSubmit}>
      <label>Nueva categoria </label>
      <input
        className="inputCategoria"
        type="text"
        value={inputValue}
        onChange={changeHandler}
        placeholder="Ingrese una categoria"
      />
    </form>
  );
};

AddCategory.propTypes = {
    addGifHandler: PropTypes.func.isRequired
}