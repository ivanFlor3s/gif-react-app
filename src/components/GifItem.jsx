import React from 'react'

export const GifItem = ({img, title}) => {
    return (
        <div className="card animate__animated  animate__fadeInLeftBig">
          <div>
              <img  className="card__img" src={img} alt={title} />
          </div>
          <div className="card__footer">
              <h4>{title}</h4>
          </div>
      </div>
    )
}   
