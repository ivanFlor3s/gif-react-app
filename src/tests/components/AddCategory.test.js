import react from "react";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Pruebas en el componente AddCAtegory", () => {
  const addGifHandler = jest.fn();
  let wrapper = shallow(
    <AddCategory addGifHandler={addGifHandler}></AddCategory>
  );

  beforeEach(() => {
    //Para limpiar mis mocks
    jest.clearAllMocks();

    wrapper = shallow(
      <AddCategory addGifHandler={addGifHandler}></AddCategory>
    );
  });

  test("Evaluo que el componente de renderize correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Simulo cambio en input", () => {
    const input = wrapper.find("input");

    //simulo que la caja de texto cambia
    input.simulate("change", { target: { value: "Hola mundo" } });
    console.log(input.props());
    // expect(input.prop('value')).toBe('Hola mundo')
    expect(input.prop("value")).toBe("");
  });

  test("No debe postear informacion con submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(addGifHandler).toHaveBeenCalled();
  });

  test("Debe de llamar setCategorias y limpiear la caja de texto", () => {
    //simulo cambio en input
    const inputHtml = wrapper.find("input");
    inputHtml.simulate("change", { target: { value: "Hola input" } });

    //simulo submit
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(addGifHandler).toHaveBeenCalled();
    expect(addGifHandler).toHaveBeenCalledWith(expect.any(Function))
    expect(inputHtml.props().value).toBe('')
  });
});
