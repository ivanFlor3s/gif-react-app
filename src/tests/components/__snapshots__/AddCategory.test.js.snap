// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCAtegory Evaluo que el componente de renderize correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Nueva categoria 
  </label>
  <input
    className="inputCategoria"
    onChange={[Function]}
    placeholder="Ingrese una categoria"
    type="text"
    value=""
  />
</form>
`;
